AppKit/CPWindow/CPWindow.j

```diff
@@ -3844,6 +4069,46 @@ var interpolate = function(fromValue, toValue, progress)
 
 @end
 
+@implementation CPWindow (TrackingArea)
+
+- (void)_addToViewsWithTrackingAreas:(CPView)aView
+{
+    [_viewsWithTrackingAreas addObject:aView];
+    
+    var tas = [aView trackingAreas];
+    
+    for (var i = 0, nb = [tas count]; i < nb; i++)
+        [self _addTrackingArea:[tas objectAtIndex:i]];
```

Comment:
Right, but I forgot to mention that this code:

```objj
if ([_trackingAreas count] == 1)
    [_window _addToViewsWithTrackingAreas:self];
else
    [_window _addTrackingArea:trackingArea];
```

shouldn't be in CPView. That logic should be in _addToViewsWithTrackingAreas, which should do nothing if the view is already in the array of tracking area views. Same goes for _removeFromViewsWithTrackingAreas.